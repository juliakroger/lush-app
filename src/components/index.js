export { default as Icon } from './Icon/Icon'
export { default as SubscriptionButton } from './SubscriptionButton/SubscriptionButton'
export { default as PaymentsAccepted } from './PaymentsAccepted/PaymentsAccepted'
