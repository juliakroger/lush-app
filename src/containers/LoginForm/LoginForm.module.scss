@import '../../scss/variables.scss';
@import '../../scss/responsive-utils';

.loginForm {
  display: flex;
  flex-direction: column;
  position: relative;
  min-height: 400px;
  width: 80%;
  max-width: 500px;
  margin: 0 auto;
  text-align: center;
  margin-top: 200px;

  input {
    margin: 0 auto;
    width: 100%;
  }
}
