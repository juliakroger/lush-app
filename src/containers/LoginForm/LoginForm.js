import React from 'react'
import PropTypes from 'prop-types'

import { Input, Button } from '../../components'
import styles from './LoginForm.module.scss'

const LoginForm = ({ email, setEmail, password, setPassword, submit }) => {
  return (
    <div className={styles.loginForm}>
      <h1>Sign In</h1>
      <Input
        placeholder='amy@example.com'
        value={email}
        valueChange={value => setEmail(value)}
        id='#signin-email'
        label='Email'
        type='email'
      />
      <Input
        value={password}
        valueChange={value => setPassword(value)}
        id='#signin-password'
        label='Password'
        secondaryLabel='Forgot password?'
        type='password'
        clickSecondary={() => {}}
      />

      <Button
        text='sign in'
        type={!email || !password ? 'buttonLight' : 'buttonMint'}
        id='#login-button'
        clickFunction={submit}
      />
      <span>
        Don&apos;t have an account? Create one <a href='/'>here</a>.
      </span>
    </div>
  )
}

LoginForm.propTypes = {
  email: PropTypes.any,
  setEmail: PropTypes.func.isRequired,
  password: PropTypes.any,
  setPassword: PropTypes.func.isRequired,
  submit: PropTypes.func.isRequired
}

export default LoginForm
