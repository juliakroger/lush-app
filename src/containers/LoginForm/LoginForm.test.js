import React from 'react'
import { shallow } from 'enzyme'
import { Button, Input } from '../../components'
import LoginForm from './LoginForm'

describe('Container - Login Form', () => {
  it('should render correctly with all props', () => {
    const component = shallow(
      <LoginForm
        email=''
        setEmail={jest.fn()}
        password=''
        setPassword={jest.fn()}
        submit={jest.fn()}
      />
    )
    expect(component).toMatchSnapshot()
  })

  it('should trigger email input change', () => {
    let email = ''

    const component = shallow(
      <LoginForm
        email={email}
        setEmail={value => (email = value)}
        password=''
        setPassword={jest.fn()}
        submit={jest.fn()}
      />
    )

    component
      .find(Input)
      .at(0)
      .props()
      .valueChange('emailtest')

    expect(email).toBe('emailtest')
  })

  it('should trigger password input change', () => {
    let password = ''

    const component = shallow(
      <LoginForm
        email=''
        setEmail={jest.fn()}
        password={password}
        setPassword={value => (password = value)}
        submit={jest.fn()}
      />
    )

    component
      .find(Input)
      .at(1)
      .props()
      .valueChange('passwordtest')

    expect(password).toBe('passwordtest')
  })

  it('should trigger submit button click', () => {
    const mockFunc = jest.fn()

    const component = shallow(
      <LoginForm
        email=''
        setEmail={jest.fn()}
        password=''
        setPassword={jest.fn()}
        submit={mockFunc}
      />
    )

    component
      .find(Button)
      .props()
      .clickFunction()
    expect(mockFunc.mock.calls.length).toBe(1)
  })
})
