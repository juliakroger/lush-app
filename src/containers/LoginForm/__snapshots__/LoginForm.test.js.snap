// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container - Login Form should render correctly with all props 1`] = `
<div
  className="loginForm"
>
  <h1>
    Sign In
  </h1>
  <Input
    blurButtonFunction={[Function]}
    id="#signin-email"
    isRequired={false}
    label="Email"
    placeholder="amy@example.com"
    secondaryLabel={null}
    type="email"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    clickSecondary={[Function]}
    id="#signin-password"
    isRequired={false}
    label="Password"
    placeholder=""
    secondaryLabel="Forgot password?"
    type="password"
    value=""
    valueChange={[Function]}
  />
  <Button
    clickFunction={[MockFunction]}
    id="#login-button"
    text="sign in"
    type="buttonLight"
  />
  <span>
    Don't have an account? Create one 
    <a
      href="/"
    >
      here
    </a>
    .
  </span>
</div>
`;
